torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -1037,17 +1037,10 @@ def step(
         .. note: Any extra parameters are passed to the base optimizer as-is.
         """
         if self._overlap_with_ddp:
-            # If DDP buckets have been rebuilt, calling `step()` indicates that
-            # the backward pass has fully completed and all information has
-            # been collected; hence, this ZeRO instance can be initialized
-            if self._overlap_info.status == _OverlapStatus.DDP_HAS_REBUILT_BUCKETS:
-                self._overlap_info.status = _OverlapStatus.INITIALIZED
-                # Since all information has been collected, perform the delayed
-                # initialization of the local optimizer and supporting state
-                self._init_zero_for_overlap()
-
-            # `step()` does not actually perform any parameter updates and is
-            # only used for bookkeeping when `overlap_with_ddp=True`
+            logging.warning(
+                "`step()` should not be included in the training loop when "
+                "`overlap_with_ddp=True`"
+            )
             return None
```

Comment:
`step()` will become a no-op just in case people still call it in the training loop. With the warning, they should know to remove it.